src/pkg/utils/component_report.go

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package utils provides generic helper functions.
+package utils
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/config/lang"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/openvex/go-vex/pkg/vex"
+)
+
+// VexType is the type of report that is a VEX document
+const VexType = "vex"
+
+// ZarfReport Extends src/types.ZarfComponentReport to add methods
+type ZarfReport struct {
+	ZarfComponentReport types.ZarfComponentReport
+}
+
+// ValidateReportSource validates vex type reports
+func (report *ZarfReport) ValidateVexReport(filePath string) error {
+	// overload method to allow for file_path to be passed in or be the report source
+	if filePath == "" {
+		filePath = report.ZarfComponentReport.Source
+	}
+	path, err := os.Stat(filePath)
+	if err != nil {
+		return fmt.Errorf(lang.PkgValidateErrPath, err)
+	}
+	if !path.IsDir() {
+		// check valid vex document
+		vexDoc, err := vex.Load(filePath)
+		if err != nil {
+			return err
+		}
+		for _, s := range vexDoc.Statements {
+			err := vex.Statement.Validate(s)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		message.Debugf("VEX path is a directory!")
+		file, err := os.Open(filePath)
+
+		if err != nil {
+			return fmt.Errorf(lang.PkgValidateErrPath, err)
+		}
+
+		defer file.Close()
+
+		files, err := file.Readdirnames(0)
+		message.Debugf("Files found are: %s", files)
+
+		if err != nil {
+			return fmt.Errorf(lang.PkgValidateErrPath, err)
+		}
+
+		for _, f := range files {
+			filePath := fmt.Sprintf("%s/%s", report.ZarfComponentReport.Source, f)
```

Comment:
no use of `/` in paths, see previous comments

Reply:
updated